thinc/legacy/loss.py

```diff
@@ -0,0 +1,268 @@
+            raise ValueError(err)
+        if xp.any(target > 1) or xp.any(target < 0):  # pragma: no cover
+            err = f"Cannot calculate CategoricalCrossentropy loss with truth values outside the [0,1] interval."
+            raise ValueError(err)
+        difference = guesses - target
+        difference *= mask
+        if self.normalize:
+            difference = difference / guesses.shape[0]
+        return difference
+
+    def get_loss(self, guesses: Floats2d, truths: IntsOrFloatsOrStrs) -> float:
+        d_truth = self.get_grad(guesses, truths)
+        return self._get_loss_from_grad(d_truth)
+
+    def _get_loss_from_grad(self, d_truth: Floats2d) -> float:
+        # TODO: Add overload for axis=None case to sum
+        return (d_truth**2).sum()  # type: ignore
+
+
+class LegacySequenceCategoricalCrossentropy(SequenceCategoricalCrossentropy):
+    def __init__(
+        self,
+        *,
+        normalize: bool = True,
+        names: Optional[Sequence[str]] = None,
+        missing_value: Optional[Union[str, int]] = None,
+        neg_prefix: Optional[str] = None,
+        label_smoothing: float = 0.0,
+    ):
+        self.cc = LegacyCategoricalCrossentropy(
+            normalize=False,
+            names=names,
+            missing_value=missing_value,
+            neg_prefix=neg_prefix,
+            label_smoothing=label_smoothing
+        )
+        self.normalize = normalize
+
+    def __call__(
+        self, guesses: Sequence[Floats2d], truths: Sequence[IntsOrFloatsOrStrs]
+    ) -> Tuple[List[Floats2d], float]:
+        grads = self.get_grad(guesses, truths)
+        loss = self._get_loss_from_grad(grads)
+        return grads, loss
+
+    def get_grad(
+        self, guesses: Sequence[Floats2d], truths: Sequence[IntsOrFloatsOrStrs]
+    ) -> List[Floats2d]:
+        err = "Cannot calculate SequenceCategoricalCrossentropy loss: guesses and truths must be same length"
+        if len(guesses) != len(truths):  # pragma: no cover
+            raise ValueError(err)
+        n = len(guesses)
+        d_scores = []
+        for yh, y in zip(guesses, truths):
+            d_yh = self.cc.get_grad(yh, y)
+            if self.normalize:
+                d_yh /= n
+            d_scores.append(d_yh)
+        return d_scores
+
+    def get_loss(
+        self, guesses: Sequence[Floats2d], truths: Sequence[IntsOrFloatsOrStrs]
+    ) -> float:
+        return self._get_loss_from_grad(self.get_grad(guesses, truths))
+
+    def _get_loss_from_grad(self, grads: Sequence[Floats2d]) -> float:
+        loss = 0.0
+        for grad in grads:
+            loss += self.cc._get_loss_from_grad(grad)  # type: ignore
+        return loss
+
+
+@registry.losses("CategoricalCrossentropy.v1")
+def configure_CategoricalCrossentropy_v1(
+    *,
+    normalize: bool = True,
+    names: Optional[Sequence[str]] = None,
+    missing_value: Optional[Union[str, int]] = None,
+) -> LegacyCategoricalCrossentropy:
+    return LegacyCategoricalCrossentropy(
```

Comment:
So the hope is that now, we can have
``` 
 -> CategoricalCrossentropy:
```
as return type here (and for the other older versions)

Reply:
ah right it does work i think!